### ISSUE TITLE
What is the digest2.txt file ?

### ISSUE THREAD

**Issue by borgogelli** [threerings/getdown#83]
What is the _digest2.txt_ file ? What are the differences with the _digest.txt_ file ? When I have to use it and when the _digest.txt_ ?

Thank you in advance

**Comment by samskivert** [comment]
Here's the email I sent when I did the release that introduced digest2.txt:

---------- Forwarded message ----------
From: Michael Bayne <mdb@samskivert.com>
Date: Sat, Nov 5, 2016 at 7:17 AM
Subject: Getdown 1.6
To: "ooo-libs@googlegroups.com" <ooo-libs@googlegroups.com>

I have released Getdown 1.6, it is making its way to Maven Central as I type.

This release introduces a new digest algorithm which fixes a few things people have requested. Specifically: things inside the META-INF directory of a jar file are included in its hash, the hash algorithm has been changed from MD5 to SHA1, and the algorithm used to sign the digest.txt file (if you're signing things) is changed from SHA1 to SHA256.

Because I cannot just change the digest algorithm without breaking old clients, this new version creates two digest files: digest.txt which uses the old algorithm and digest2.txt which uses the new algorithm. This ensures that when you ship an app built with this new Getdown, old versions of Getdown can still understand and upgrade your app. They will use the digest.txt file which is exactly as it has always been.

Assuming you have a process in place for Getdown to update itself inside your app's installation, then once the user has installed your update which was built with the new Getdown, your app will start using the new Getdown which will use the digest2.txt file going forward.

Since these files are small, we'll probably just generate both digest.txt files forever, because that means that someone who hasn't updated their app in a really long time will still be able to make this transition at any point in the future, but if there is demand, I will add a way to turn off the old digest.txt generation for people who know they're only ever going to have clients running the new Getdown.

The Ant task now generates both digests automatically, but the Maven plugin needs to be updated to call the new function that generates all the digests instead of just one.

This release also includes another pull request which ensures that non-versioned installations download update files to a separate location and only install them after validating that they've been completely and successfully downloaded. So they won't cause problems if the download fails halfway. This was never a problem for versioned installations, but a lot of people seem to be using non-versioned installations in production (which was never the intent, but hey apparently it's useful).

Another pull request is included which introduces a use_code_cache configuration parameter. If you set it to true, Getdown will copy all of the code files into a temporary directory before launching the application and set the classpath to reference those temporary files. This allows the original application to be updated even if old instances of the application are still running.

There are a lot of other small tweaks and fixes, check the commit history for all the details:

https://github.com/threerings/getdown/commits/master

-- mdb@samskivert.com
